src/Chainweb/Pact/Types.hs

```diff
@@ -140,7 +140,7 @@ data PactDbStatePersist = PactDbStatePersist
     , _pdbspPactDbState :: PactDbState
     }
 
-newtype GasSupply = GasSupply { _gasSupply :: Decimal }
+newtype GasSupply = GasSupply { _gasSupply :: GasPrice }
```

Comment:
`GasSupply` is our current working total. Why was `Decimal` replaced with `GasPrice`?

Reply:
Maybe not the best idea, it was in hopes of minimizing coercion. I'll look at this

Reply:
Ah the other reason is it needs a JSON representation to go into the message body, ie in `mkBuyGasCmd` and `GasPrice` has the right instances.

The larger issue is using dedicated newtypes for gas prices, when in reality these are simply money values which don't have a first class representation in the API. I propose leaving this for now and addressing in a Pact PR, ie a `Price` or `Money` API type.

Reply:
OK figured out a better solution, and one we should have done for `GasPrice`: newtype `ParsedDecimal` to grab its instances. Likewise, `GasLimit` should newtype `ParsedInteger`.

Reply:
Works for me